design-proposals/eim-standalone-managed-en.md

```diff
@@ -0,0 +1,236 @@
+
+The current design assumes that the user will SSH into a Standalone EN and trigger onboarding one by one (it can still be automated by a script).
+Depending on the customers' requirements we can provide a kind of a "Bulk Onboarding Tool" that will allow to onboard multiple ENs at once.
+
+### How to map local OS users to Local Accounts?
+
+For now, we won't import existing OS users as local accounts. If we support NIO in the future,
+we can add support for defining local accounts that should be configured on the pre-provisioned SEN that is onboarded.
+
+## Affected components and Teams
+
+### Impact on OS profiles
+
+The design assumes that the same OS profile, that is used by SEN, is already created in the Edge Orchestrator.
+Therefore, we will provide a user guide on how to create OS profiles before scaling out SEN deployment. No UI support is planned for now.
+
+However, this can have the following limitations: 1) OS profile must always be supported by the Edge Orchestrator,
+but this can be a customized OS image that is not supported by the Edge Orchestrator,
+2) it requires additional step for users to go and create OS profiles in the Edge Orchestrator before onboarding (if they don't exist).
+
+See open issues for more considerations on this topic.
+
+### Impact on Multi-Tenancy and IAM
+
+There should be a dedicated IAM role that is only allowed to onboard standalone Edge Nodes to the Edge Orchestrator.
+It should be a different role than the current onboarding role that is used for remote onboarding.
+
+### Impact on the Edge Microvisor Toolkit
+
+The design will require a new CLI tool that will be baked into the EMT image.
+
+## Implementation plan
+
+The implementation targets 3.1. The implemention will include:
+- Modifications to the Onboarding Manager to support onboarding of Standalone ENs
+- New CLI tool that is developed as part of infra-onboarding or integrated with EMF CLI tool
+- New SPEC file to the Edge Microvisor Toolkit to bake the new CLI tool into the OS image
+
+The implementation will be done by the EIM team.
+
+## Open issues (if applicable)
+
+- The OS version of Standalone ENs deployed by customers may not have a corresponding OS Profile in the Edge Orchestrator.
+  For instance, the customer may have deployed a Standalone EN with custom EMT image that never existed in the Edge Orchestrator.
+  Another example is a customer that has deployed a Standalone EN with an old OS image that is not supported by the Edge Orchestrator 
+  (which has already been upgraded to use newer versions). Yet another issue is that the OS version from `/etc/os-release`
+  will not uniquely identify the OS image if that's a mutable OS (Ubuntu case) and Onboarding Manager is unable to query or validate mutable OS profile
+  based on info provided from SENs.
+  There are possible solutions to this:
+  - Users should be able to create their own OS Profiles that uses a custom OS image that was used for Standalone ENs.
```

Comment:
shall we put some guardrails in the OS_profile ? For example prevent the provisioning or a/b updates based on the type of the profile

Reply:
maybe add STANDALONE_OS as type and ofrce the user to use that